packages/firebase_remote_config/ios/Classes/FirebaseRemoteConfigPlugin.m

```diff
@@ -97,12 +104,28 @@ - (void)handleMethodCall:(FlutterMethodCall *)call result:(FlutterResult)result
                     }
                   }];
   } else if ([@"RemoteConfig#activate" isEqualToString:call.method]) {
-    BOOL newConfig = [[FIRRemoteConfig remoteConfig] activateFetched];
-    NSDictionary *parameters = [self getConfigParameters];
-    NSMutableDictionary *resultDict = [[NSMutableDictionary alloc] init];
-    resultDict[@"newConfig"] = [NSNumber numberWithBool:newConfig];
-    resultDict[@"parameters"] = parameters;
-    result(resultDict);
+    [[FIRRemoteConfig remoteConfig]
+        activateWithCompletion:^(BOOL changed, NSError *_Nullable error) {
+          BOOL newConfig = @true;
+          if (error) {
+            if (error.userInfo && error.userInfo[@"ActivationFailureReason"] != nil &&
+                [error.userInfo[@"ActivationFailureReason"] containsString:@"already activated"]) {
+              newConfig = @false;
```

Comment:
should this `newConfig ` assignment be in the outer `if` block? assuming both errors cause config activation to fail?

Reply:
Interesting - will spiff this area up - you are right that is safer, however the else block calls `result(...)` which means (I think?) that the newConfig value will never be used as the statements where it would be assigned into a results dict and returned won't be hit. So I don't think there was an error currently but it's not great if that's not immediately clear on a read

Structurally for max safety, seems like I should start with it set to `@false` really, and then after the error handling block set it to `@true` with a little `// no errors, we have a new config` side comment

Reply:
I'm having a look at the RNFB implementation, and I see it mirrors that. To my mind, it doesn't seem right to return two different types (i.e an error and a successful resolution albeit with activated as `false`) for an error. But I'm more than likely missing something so I think this can be safely ignored 😄

Reply:
Yeah - you will likely not be surprised to learn that I lifted this from the RNFB implementation :-), I agree it seems a little odd however when I took a little stab at doing it differently it meant changing up how the Dart code above this native layer worked since it already had the "newConfig" native toggle style and used the results dictionary to cache the values for later retrieval. It wasn't a huge change but meant the PR was less surgical then I wanted given the real point was just to ingest SDK API changes, not clean up how it worked. So I left it